### PR TITLE
Improvements to Analytics

### PR DESCRIPTION
**Pull request by ruler501**
Allow prerendering without doing any calculations and add a tagCount analytic.

### CODE REVIEW

File: routes/cube_routes.js [hunk 1]
```diff
@@ -737,8 +739,9 @@ router.get('/list/:id', async (req, res) => {
       cardNames.push(card.details.name);
     });
 
-    const priceDict = await GetPrices([...pids]);
-    const eloDict = await getElo(cardNames, true);
+    const priceDictQ = GetPrices([...pids]);
```
Comment (dekkerglen) [inline routes/cube_routes.js]: we should just pass `pids` into here

File: routes/cube_routes.js [hunk 2]
```diff
@@ -850,12 +853,14 @@ router.get('/playtest/:id', async (req, res) => {
 
 router.get('/analysis/:id', async (req, res) => {
   try {
-    const cube = await Cube.findOne(buildIdQuery(req.params.id)).lean();
+    const fields = 'cards name owner defaultDraftFormat draft_formats card_count type';
+    const cube = await Cube.findOne(buildIdQuery(req.params.id), fields).lean();
```
Comment (dekkerglen) [inline routes/cube_routes.js]: why create a variable here, would be cleaner to just pass the string in directly

File: routes/tools_routes.js [hunk 3]
```diff
@@ -58,7 +58,8 @@ async function matchingCards(filter) {
       cards = Filter.filterCardsDetails(cards, filter);
     }
     if (Filter.filterUses(filter, 'price') || Filter.filterUses(filter, 'price_foil')) {
-      const withPrices = await addPrices(cards);
+      const withPricesPre = await addPrices(cards.map((card) => ({ details: card })));
```
Comment (dekkerglen) [inline routes/tools_routes.js]: why are you doing this

Comment (ruler501) [reply routes/tools_routes.js]: If we want to be able to use addPrices with cube card objects it needs to take them this way.

File: routes/cube_routes.js (hunk 1 above)
Comment (ruler501) [reply routes/cube_routes.js]: I believe that didn't work because pids is a Set and GetPrices only works with an array.

**Comment by phulin** [comment]
Can you try and separate these into smaller PRs in the future? Very hard to review something with this many moving parts.

**Comment by ruler501** [comment]
Sure

**Comment by dekkerglen** [comment]
Conflicts

**Pull request closed without merging**